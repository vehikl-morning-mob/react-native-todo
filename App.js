import React from 'react';
import {
  Button,
  StyleSheet,
  Text,
  TextInput,
  View,
  TouchableHighlight,
  FlatList
} from 'react-native';
import Task from './components/Task';

const tasks = [
  {
    id: '',
    name: '',
    description: '',
    completed: false,
  }
];

const createTask = ({ id, name }) => {
  return {
    id,
    name,
    completed: false,
  }
}

export default class App extends React.Component {
  state = {
    tasks: [],
    taskName: '',
    showCompleted: true,
  }

  addTask = () => {
    const { taskName, tasks } = this.state;

    if (!taskName.trim()) return;

    const newTask = {
      id: Math.random().toString(32),
      name: taskName,
    }

    this.setState({
      tasks: tasks.concat(createTask(newTask)),
      taskName: ''
    })
  }

  handleInput = (value) => {
    this.setState({ taskName: value });
  }

  toggleCompleted = (task) =>  {
    this.setState({
      tasks: this.state.tasks.map(t => ({
        ...t,
        completed: t.id === task.id ? !t.completed : t.completed,
      })),
    })
  }

  toggleShowCompleted = () => {
    this.setState({
      showCompleted: !this.state.showCompleted
    })
  }

  renderTask = ({item}) => (
    <Task task={item} toggleCompleted={this.toggleCompleted} key={item.id}></Task>
  )
  render() {
    const tasks = this.state.showCompleted
      ? this.state.tasks
      : this.state.tasks.filter(task => !task.completed)

    return (
      <View style={styles.container}>
        <TextInput
          style={styles.input}
          value={this.state.taskName}
          onChangeText={this.handleInput}
          onSubmitEditing={this.addTask}
          blurOnSubmit={false}
        />
        <TouchableHighlight onPress={this.addTask}>
          <Text>Add task</Text>
        </TouchableHighlight>
        <FlatList data={tasks} renderItem={this.renderTask} keyExtractor={(item, index) => item.id } />
        <Button title={`${this.state.showCompleted ? 'Hide': 'Show'} Completed`} onPress={this.toggleShowCompleted}/>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
    marginTop: 100,
  },

  input: {
    borderWidth: 1,
    borderColor: 'black',
    width: '75%',
  }
});
