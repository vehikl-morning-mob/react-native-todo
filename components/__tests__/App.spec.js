import React from 'react';
import renderer from 'react-test-renderer';
import App from '../../App'
import { TextInput, TouchableHighlight } from 'react-native';
import TaskEntity from '../../entities/task';


describe('App', () => {
  describe('Incoming Commands', () => {
    it('can hide completed tasks', () => {
      const testInstance = renderer.create(<App />);
      const component = testInstance.root.instance;
      const task = new TaskEntity(1, 'whatever man', true)

      component.setState({
        tasks: [
          task
        ],
        showCompleted: false
      })
      expect(testInstance.toJSON()).toMatchSnapshot();
    })



    it('Adds task using add task touchableHighlight', () => {
      const component = renderer.create(<App />);
      const instance = component.root;
      instance.findByType(TextInput).props.onChangeText('task1');
      instance.findByType(TouchableHighlight).props.onPress();
      expect(component.toJSON()).toMatchSnapshot();
    })

    it('Adds a task when enter/return is pressed', () => {
      const component = renderer.create(<App />);
      const instance = component.root;

      instance.findByType(TextInput).props.onChangeText('task1');
      instance.findByType(TextInput).props.onSubmitEditing();
      expect(component.toJSON()).toMatchSnapshot();
    })


  })

  describe('Incoming Queries', () => {
    it('Renders an empty list', () => {
      const component = renderer.create(<App />).toJSON();
      expect(component).toMatchSnapshot();
    })
  })
})