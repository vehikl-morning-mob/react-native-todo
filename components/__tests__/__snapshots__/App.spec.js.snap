// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Adds a task when enter/return is pressed 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "marginTop": 100,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    blurOnSubmit={false}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    style={
      Object {
        "borderColor": "black",
        "borderWidth": 1,
        "width": "75%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <Text
      style={null}
    >
      Add task
    </Text>
  </View>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Hide Completed
      </Text>
    </View>
  </View>
</View>
`;

exports[`App Adds task using add task touchableHighlight 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "marginTop": 100,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    blurOnSubmit={false}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    style={
      Object {
        "borderColor": "black",
        "borderWidth": 1,
        "width": "75%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <Text
      style={null}
    >
      Add task
    </Text>
  </View>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Hide Completed
      </Text>
    </View>
  </View>
</View>
`;

exports[`App Incoming Commands can hide completed tasks 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "marginTop": 100,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    blurOnSubmit={false}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    style={
      Object {
        "borderColor": "black",
        "borderWidth": 1,
        "width": "75%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <Text
      style={null}
    >
      Add task
    </Text>
  </View>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Show Completed
      </Text>
    </View>
  </View>
</View>
`;

exports[`App Renders an empty list 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "marginTop": 100,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    blurOnSubmit={false}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    style={
      Object {
        "borderColor": "black",
        "borderWidth": 1,
        "width": "75%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <Text
      style={null}
    >
      Add task
    </Text>
  </View>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Hide Completed
      </Text>
    </View>
  </View>
</View>
`;
